sentry/src/main/java/io/sentry/Span.java

```diff
@@ -317,4 +366,42 @@ public boolean isNoOp() {
   void setSpanFinishedCallback(final @Nullable SpanFinishedCallback callback) {
     this.spanFinishedCallback = callback;
   }
+
+  public void setStartDate(@NotNull SentryDate date) {
+    this.startTimestamp = date;
+  }
+
+  /**
+   * Updates the end date of the span. Note: This will only update the end date if the span is
+   * already finished.
+   *
+   * @param date the end date to set
+   * @return true if the end date was updated, false otherwise
+   */
+  public boolean updateEndDate(@NotNull SentryDate date) {
+    if (this.timestamp != null) {
+      this.timestamp = date;
+      return true;
+    }
+    return false;
+  }
+
+  @NotNull
+  public SpanOptions getOptions() {
```

Comment:
`m`: maybe we should make this method package-private? Don't think we need to expose it
```suggestion
  SpanOptions getOptions() {
```